app/models/post.rb

```diff
@@ -0,0 +1,11 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  after_save :update_post_counter
+  # Custom Method: Update the posts counter for a user
+  def update_post_counter
+    author.increment!(:posts_counter)
+  end
+end
```

Comment:
- [ ] Great job so far. It is required to have a method in the post model that returns thhe most recent comments for a given post. That has not been implemented yet. Kindly add that method too.

### Project Requirement.

![post-comments](https://github.com/geekelo/rails-blog_app/assets/26597432/c256cd1b-bd6c-43b0-9d42-3d5781d14bce)

- [OPTIONAL] It would be nice to make your `update_post_method` a private method so as to help enforce encapsulation which is one of the fundamental principles of object-oriented programming. That means object's internal state and implementation details are hidden from the outside world. By making certain methods private, you can hide the implementation details and only expose the public interface of the class. This makes the code more maintainable and less prone to external interference.
You can check [this article](https://www.rubyguides.com/2018/10/method-visibility/) for more details